analytics/analytics.go

```diff
@@ -158,51 +147,103 @@ func (dc *dataCollection) update() {
 	dc.NumPeers = uint64(len(st.Peers))
 }
 
+func (dc *dataCollection) getGrpcConn() (*grpc.ClientConn, context.CancelFunc, error) {
+	config, err := dc.node.Repo.Config()
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to load config: %s", err.Error())
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	conn, err := grpc.DialContext(ctx, config.StatusServerDomain, grpc.WithInsecure(), grpc.WithDisableRetry())
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to connect to status server: %s", err.Error())
+	}
+	return conn, cancel, nil
+}
+
 func (dc *dataCollection) sendData() {
+	retry(func() error {
+		return dc.doSendData()
+	})
+}
+
+func (dc *dataCollection) doSendData() error {
 	dc.update()
-	dcMarshal, err := json.Marshal(dc)
+	payload, err := dc.getPayload()
 	if err != nil {
 		dc.reportHealthAlert(fmt.Sprintf("failed to marshal dataCollection object to a byte array: %s", err.Error()))
-		return
+		return err
 	}
 	if dc.node.PrivateKey == nil {
 		dc.reportHealthAlert("node's private key is null")
-		return
+		return err
 	}
-	signature, err := dc.node.PrivateKey.Sign(dcMarshal)
+
+	signature, err := dc.node.PrivateKey.Sign(payload)
 	if err != nil {
 		dc.reportHealthAlert(fmt.Sprintf("failed to sign raw data with node private key: %s", err.Error()))
-		return
+		return err
 	}
+
 	publicKey, err := ic.MarshalPublicKey(dc.node.PrivateKey.GetPublic())
 	if err != nil {
 		dc.reportHealthAlert(fmt.Sprintf("failed to marshal node public key: %s", err.Error()))
-		return
+		return err
 	}
-	dataBagInstance := new(dataBag)
-	dataBagInstance.PublicKey = publicKey
-	dataBagInstance.Signature = signature
-	dataBagInstance.Payload = dcMarshal
-	dataBagMarshaled, err := json.Marshal(dataBagInstance)
+
+	sm := new(pb.SignedMetrics)
+	sm.Payload = payload
+	sm.Signature = signature
+	sm.PublicKey = publicKey
+
+	conn, cancel, err := dc.getGrpcConn()
 	if err != nil {
-		dc.reportHealthAlert(fmt.Sprintf("failed to marshal databag: %s", err.Error()))
-		return
+		return err
 	}
+	defer cancel()
+	defer conn.Close()
 
-	// btfs node reports to status server by making HTTP request
-	req, err := http.NewRequest("POST", fmt.Sprintf("%s%s", statusServerDomain, routeMetrics), bytes.NewReader(dataBagMarshaled))
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```

Comment:
same comment here: "I'd make 5*time.Second a timeout const for future convenient adjustment and avoid the magic number"